frontend/src/Components/WmsSecuritySettings/WmsSecuritySettings.tsx

```diff
@@ -0,0 +1,83 @@
+import { SyncOutlined } from '@ant-design/icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { MapContext as ReactGeoMapContext } from '@terrestris/react-geo';
+import Collection from 'ol/Collection';
+import React, { ReactElement, useEffect, useState } from 'react';
+import { useParams } from 'react-router';
+import WmsRepo from '../../Repos/WmsRepo';
+import { TreeUtils } from '../../Utils/TreeUtils';
+import { TreeNodeType } from '../Shared/FormFields/TreeFormField/TreeFormFieldTypes';
+import { olMap, TheMap } from '../TheMap/TheMap';
+import { RulesDrawer } from './RulesDrawer';
+import './WmsSecuritySettings.css';
+
+const wmsRepo = new WmsRepo();
+const treeUtils = new TreeUtils();
+
+export const WmsSecuritySettings = (): ReactElement => {
+
+  // get the ID parameter from the url
+  const { id } = useParams();
+
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [initLayerTreeData, setInitLayerTreeData] = useState<Collection<any>>(new Collection());
+
+  useEffect(() => {
+    // TODO: need to add some sort of loading until the values are fetched
+    // olMap.addLayer(mapContextLayersPreviewGroup);
+    if (id) {
+      setIsLoading(true);
+      const fetchMapContext = async () => {
```

Comment:
here this method name should be changed to a more generic name, since now is used by The Map Context and the WmsSecurity Settings. But we can decide on it later